Using @since (related to WI-16509)

boekkooi opened the pull request:
This PR adds the [`@since`](http://www.phpdoc.org/docs/latest/references/phpdoc/tags/since.html) to the constant, function, define, class, variable, method where the information was already availible.

This PR relates to the comments made in [WI-16509](https://youtrack.jetbrains.com/issue/WI-16509)

The following regex and searches where used:
```
\* Available since PHP ([0-9\.]+)
\* Since PHP ([0-9\.]+)
\* PHP >= ([0-9\.]+)<br\/>
\* \(PHP [0-9] &gt;= ([0-9\.]+)\)<br\/>
``` 
(`* @since $1`)
`\* \(PHP 4 &gt;= 4\.([0-9\.]+), PHP 5\)<br\/>` (`* @since 4.$1, 5.0`)
`* (PHP 4, PHP 5)<br/>` (`* @since 4.0, 5.0`)
`* (PHP 5)<br/>` (`* @since 5.0`)
`(\s*\*\s*)@since ([0-9\.]+), ([0-9\.])` (`$1@since $2$1@since $3`)

neuro159 commented:
Impressive work! I'll pull this in 9.0.x after additional checks.
Will also work on language-level dependent checks later..
We need to have accompanying @deprecated VERSION tags too...

boekkooi commented:
@neuro159 So I added some new `@deprecated` annotations and fixed some others to be according to http://www.phpdoc.org/docs/latest/references/phpdoc/tags/deprecated.html 
There is one issue left and that is **removed** method/functions/classes for example [mysqli::client_encoding](http://php.net/manual/en/function.mysqli-client-encoding.php) which was removed in PHP 5.4. Should I add a `@removed  [<version>] [<description>]` tags? This would be a PHPStorm specific tag but I think it is very much needed to fix WI-16509

neuro159 commented:
Thank you. No need for another custom tag.
The work saving both @since & @deprecated versions in index and taking into account on inspection, completion, etc. is still pending, ETA v10 EAP
